ui/plugins/eu.esdihumboldt.hale.ui/src/eu/esdihumboldt/hale/ui/service/project/internal/ProjectVariablesPreferencePage.java

```diff
@@ -205,6 +205,11 @@ public void widgetSelected(SelectionEvent e) {
 						c.dispose();
 					}
 					variables.remove(varName);
+
+					ProjectService ps = PlatformUI.getWorkbench().getService(ProjectService.class);
+					ps.getConfigurationService().setProperty(
+							ProjectVariables.PROJECT_PROPERTY_VARIABLES, Value.complex(variables));
```

Comment:
This will directly store the removed variable in the project service, when a user clicked on remove. This should only happen though when a user actually applies the changes from the preference page.

To me it looks like what is missing here could be a `varControls.remove(varName);`.